### PR TITLE
updated for chef 14 and added windows support

### PR DESCRIPTION
**Pull request by Stromweld**

### CODE REVIEW

**Comment by Stromweld** [comment]
I believe this pull request fixes #45 #48 #50 by using the built-in chef_handler resource now in chef-client 14+ and removing the need for the now deprecated chef_handler cookbook. By also using the chef cache path makes the cookbook universal with most operating systems.

**Comment by Stromweld** [comment]
@martinb3 what do you think of this update?

**Comment by marcencov** [comment]
@martinb3 Please apply fixes for chef 14.

**Comment by mateuszkwiatkowski** [comment]
I confirm that cookbook with these changes works. Please merge this PR and release a new version to the supermarket. Thank you!

**Comment by Stromweld** [comment]
@shortdudey123 can you review this?

File: recipes/default.rb [hunk 1]
```diff
@@ -41,14 +41,14 @@
   chef_gem 'slackr' do
     compile_time false if respond_to?(:compile_time)
   end
-  handler_file = "#{node['chef_handler']['handler_path']}/slack_handler.rb"
```
Comment (martinb3) [inline recipes/default.rb:44]: If we're dropping chef_handler, do we also need to remove it from the dependencies in metadata.rb and stop including it in recipes?

Comment (Stromweld) [reply recipes/default.rb:44]: chef_handler is now a built-in resource in chef 14+ so there is no attribute for handler_path anymore as the chef_handler cookbook is deprecated and no longer recommended for use. That's also why I put in the major version bump in metadata. https://supermarket.chef.io/cookbooks/chef_handler

Comment (Stromweld) [reply recipes/default.rb:44]: so to answer your question yes.

Comment (Stromweld) [reply recipes/default.rb:44]: I'd recommend doing a version 0.9.2 verision bump to master with the changes you just merged in and release it, and then we can remove them from this commit and you can do a major 1.0.0 version bump and release. Users on older chef clients can then pin to ~>0.9 and users on chef 14+ can use latest 1.0.0.

**Comment by Stromweld** [comment]
@martinb3 ok this PR should be ready to go now.

**Comment by mateuszkwiatkowski** [comment]
Thank you!

**Pull request merged**